egs/librispeech/ASR/finetune_hubert_transducer/hubert_encoder.py

```diff
@@ -0,0 +1,260 @@
+        self.encoders.feature_grad_mult = 0.0
+
+        self.freeze_finetune_updates = freeze_finetune_updates
+        self.register_buffer("num_updates", torch.LongTensor([0]))
+
+        if subsample_output:
+            self.subsample = get_subsample_module(
+                subsample_mode,
+                input_size=self.hubert_output_size,
+                output_size=output_size,
+            )
+            self.subsample_mode = subsample_mode
+            logging.info("Subsample output!}")
+            logging.info(self.subsample)
+        else:
+            self.subsample = None
+            self.subsample_mode = None
+            logging.info("Do not subsample output")
+
+        self.training = training
+        delattr(self.encoders, "final_proj")
+        delattr(self.encoders, "label_embs_concat")
+
+    @classmethod
+    def add_arguments(cls, parser):
+        parser.add_argument(
+            "--hubert-model-dir",
+            type=str,
+            help="Path to the pretrained Hubert model",
+        )
+
+        parser.add_argument(
+            "--hubert-freeze-finetune-updates",
+            type=int,
+            default=0,
+            help="The number of updates during which the transformer blocks \
+                in hubert are frozen.",
+        )
+
+        parser.add_argument(
+            "--hubert-mask-prob",
+            type=float,
+            default=0.65,
+            help="Mask probability",
+        )
+
+        parser.add_argument(
+            "--hubert-mask-channel-prob",
+            type=float,
+            default=0.5,
+            help="Mask channel probability",
+        )
+
+        parser.add_argument(
+            "--hubert-mask-channel-length",
+            type=int,
+            default=64,
+            help="Mask channel length",
+        )
+
+        parser.add_argument(
+            "--hubert-subsample-output",
+            type=str2bool,
+            default=False,
+            help="Whether subsample the hubert output to reduce frame rate",
+        )
+
+        parser.add_argument(
+            "--hubert-subsample-mode",
+            type=str,
+            default="concat_tanh",
+            choices=["concat", "concat_relu", "concat_tanh", "avgpooling"],
+        )
+
+    def normalise_input(
+        self, x: torch.Tensor, x_lens: torch.Tensor
+    ) -> torch.Tensor:
+        """Normalize a batch of audio samples to zero mean unit variance
+
+        Args:
+            x (torch.Tensor): a batch of audio samples
+            x_lens (torch.Tensor): # audio samples
+
+        Returns:
+            torch.Tensor: normalized audio samples
+        """
+        row_mean = (x.sum(dim=-1) / x_lens).view(x.size(0), 1)
+        row_std = (
+            torch.tensor([sample[sample != 0.0].std() for sample in x])
+            .view(x.size(0), 1)
+            .to(x.device)
+        )
+        for i in range(x.size(0)):
+            x[i, : x_lens[i]] = (x[i, : x_lens[i]] - row_mean[i]) / row_std[i]
```

Comment:
Shall we remove `row_std` and replace this line with
```suggestion
            x[i, : x_lens[i]] = (x[i, : x_lens[i]] - row_mean[i]) / x[i, : x_lens[i]].std()
```